### PR TITLE
Update `azurerm_policy_set_definition` - Fix updates for `parameters` and `parameter_values`

### LINKED ISSUE #8880: `azurerm_policy_set_definition` reports error during update
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

terraform 0.12.29

### Affected Resource(s)

* `azurerm_policy_set_definition`

### Terraform Configuration Files

```hcl
provider "azurerm" {
  # whilst the `version` attribute is optional, we recommend pinning to a given version of the Provider
  version = ">=2.31.0"
  features {}
}
data "azurerm_policy_definition" "activity_log_exist_admin_operations" {
  display_name = "An activity log alert should exist for specific Administrative operations"
}
resource "azurerm_policy_set_definition" "cis_custom" {
  name                  = "CIS Custom Effects"
  policy_type           = "Custom"
  display_name          = "CIS Custom Effects"
  policy_definition_reference {
    policy_definition_id = data.azurerm_policy_definition.activity_log_exist_admin_operations.id
    parameter_values = jsonencode({
      operationName = { value = "Microsoft.Sql/servers/firewallRules/delete" }
    })
  }
}
```

### Expected Behavior

The policy set definition could be updated successfully

### Actual Behavior

An error occurs:
```
Error: expanding `policy_definition_reference`: cannot set both `parameters` and `parameter_values`
```

### Steps to Reproduce

1. `terraform apply` with the config above
1. change the display name and `terraform apply` again

### References

* #0000

### PR DESCRIPTION
**Pull request by ArcturusZhang**
Fixes #8880

### CODE REVIEW

**Review by jackofallops — COMMENTED** [review]
Hi @ArcturusZhang - Thanks for this PR, Looking good so far.  I've made a suggestion on one item below, and there's a couple lint/test issues in travis  to look at.  I'll loop back once you've updated for those to continue review.

File: azurerm/internal/services/policy/policy_set_definition_resource.go [hunk 1]
```diff
@@ -316,6 +313,107 @@ func resourceArmPolicySetDefinitionCreateUpdate(d *schema.ResourceData, meta int
...
+	id, err := parse.PolicySetDefinitionID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	managementGroupName := ""
+	switch scopeId := id.PolicyScopeId.(type) { // nolint gocritic
+	case parse.ScopeAtManagementGroup:
+		managementGroupName = scopeId.ManagementGroupName
+	}
```
[the remark this answers is not in the document]
Comment (ArcturusZhang) [reply azurerm/internal/services/policy/policy_set_definition_resource.go:330]: Make sense. And the same pattern also exist in the `Get` callback, I would like to change that too.

**Comment by ArcturusZhang** [comment]
Hi @jackofallops I make some changes, please have a look, thanks

**Comment by ArcturusZhang** [comment]
Hi @jackofallops I also change the `JSON object` statement in docs to `JSON string` to reduce misleading (for instance reported in [this issue](https://github.com/terraform-providers/terraform-provider-azurerm/issues/8880))

**Review by jackofallops — APPROVED** [review]
Thanks @ArcturusZhang - LGTM now 👍

**Comment by jackofallops** [comment]
Tests Passing (2 failures unrelated)
image

**Pull request merged**